### ISSUE TITLE
Multi loss support

### ISSUE THREAD

**Issue by penguinmenac3** [penguinmenac3/deeplearning-starterpack#1]
Make switching between multiple losses and comparing results easy.

Suggestion switch create loss pattern to be different. (like below)
Possibly even add multi optimizer support

```python
def create_loss(hyper_params, train_model, validation_model, train_labels, validation_labels=None):
    reports = []
    loss_ops = {}
    validation_loss_ops = {}
    train_labels = tf.reshape(train_labels, [-1, 1000])
    loss_ops["cross_entropy"] = tf.reduce_mean(tf.nn.softmax_cross_entropy_with_logits(logits=train_model["logits"], labels=train_labels))
    if hyper_params.train.loss not in loss_ops.keys():
        raise RuntimeException("Unsupported Loss: " + hyper_params.train.loss)
    if hyper_params.train.optimizer == "RMSProp":
        train_op = tf.train.RMSPropOptimizer(learning_rate=hyper_params.train.learning_rate,
                                             decay=hyper_params.train.decay).minimize(loss_ops[hyper_params.train.loss])
    else:
        raise RuntimeException("Unsupported Optimizer: " + hyper_params.train.optimizer)
    for loss_name in loss_ops.keys():
        tf.summary.scalar('train/' + loss_name, loss_ops[loss_name])
    reports.append(loss_ops[loss_name])

    # Create a validation loss if possible.
    if validation_labels is not None:
        validation_labels = tf.reshape(validation_labels, [-1, 1000])
        validation_loss_ops["cross_entropy"] = tf.reduce_mean(
            tf.nn.softmax_cross_entropy_with_logits(logits=validation_model["logits"], labels=validation_labels))
        for loss_name in loss_ops.keys():
            tf.summary.scalar('validation/' + loss_name, validation_loss_ops[loss_name])
        reports.append(loss_ops[loss_name])

    return train_op, reports
```

In hyperparameters json add fields for train.loss and train.optimizer.

**Comment by penguinmenac3** [comment]
with new sample of how loss should work in https://github.com/penguinmenac3/deeplearning-starterpack/blob/master/README.md#simple-to-use-tensorflow
this should be resolved. Since create loss returns a list of losses and the optimizer is extra.

**Comment by penguinmenac3** [comment]
The loss implementation is the following and only returns one loss operation and a list of metrics.

```python
def create_loss(model, labels, mode, hyper_params):
    return loss_op, metrics
```

However, multiloss can be implemented by calling multiple different losses (creating them) and then merging them)

```python
loss_1 = create_loss(...)
loss_2 = create_loss(...)
optimizer_op = some_fancy_optimizer(loss_1, loss_2)
```
